streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```diff
@@ -346,6 +346,23 @@ public static <V> Function<byte[], V> getDeserializeValue(final StateSerdes<?, V
         return byteArray -> deserializer.deserialize(serdes.topic(), byteArray);
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public static <V> Function<byte[], V> getDeserializeValue2(final StateSerdes<?, V> serdes,
+                                                               final StateStore wrapped,
+                                                               final boolean isDSLStore ) {
+        final Serde<V> valueSerde = serdes.valueSerde();
+        final boolean timestamped = WrappedStateStore.isTimestamped(wrapped) || isDSLStore;
+        final Deserializer<V> deserializer;
+        if (!timestamped && valueSerde instanceof ValueAndTimestampSerde) {
```

Comment:
We did discuss this part of the code in person, and we where both confused why it's needed -- I think it's needed because of the underlying "bug" that the `KeyValueToTimestampedKeyValueByteStoreAdapter` did not add the missing timestamp on the query result and is not recognized as a `TimestampedBytestStore` -- after this is fixed, I believe we might not need this code any longer (did not try it out, but we should explore it as side cleanup).

Reply:
Yes, I find that bug, that bug in `WrappedStateStore`,  we should add a statement like this
```
public static boolean isTimestamped(final StateStore stateStore) {
        if (stateStore instanceof KeyValueToTimestampedKeyValueByteStoreAdapter) {
            return true;
        }
        if (stateStore instanceof TimestampedBytesStore) {
            return true;
        } else if (stateStore instanceof WrappedStateStore) {

```

But this just solve the ROCK_KV DSL store with cache = true.

We have to force` timestamped` to be true.

Reply:
This just solve the KeyQuery temporary, now I am debugging at RangeQuery, we should transfer the `RocksDBRangeIterator `to `KeyValueToTimestampedKeyValueIteratorAdapter` in `KeyValueToTimestampedKeyValueByteStoreAdapter`. Because we want to transfer kv iterator to ts kv iterator.

Reply:
As discussed in person: for `RangeQuery`, `KeyValueToTimestampedKeyValueIteratorAdapter` need to wrap the provided `RocksDBRangeIterator` to translate between plain-byte[] format what we receive from the inner store, to the required timestamped-byte[] format that the upper layer expect.

Reply:
ok